mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -694,6 +771,41 @@ func resourceMongoDBAtlasClusterUpdate(d *schema.ResourceData, meta interface{})
 	if d.HasChange("auto_scaling_disk_gb_enabled") {
 		cluster.AutoScaling.DiskGBEnabled = pointy.Bool(d.Get("auto_scaling_disk_gb_enabled").(bool))
 	}
+	if d.HasChange("auto_scaling_compute_enabled") {
+		compute.Enabled = pointy.Bool(d.Get("auto_scaling_compute_enabled").(bool))
+		autoScaling.Compute = compute
+		cluster.AutoScaling = autoScaling
+	}
+	if d.HasChange("auto_scaling_compute_scale_down_enabled") {
+		compute.ScaleDownEnabled = pointy.Bool(d.Get("auto_scaling_compute_scale_down_enabled").(bool))
+		autoScaling.Compute = compute
+		cluster.AutoScaling = autoScaling
+	}
+	providerAutoScaling := &matlas.AutoScaling{}
+	compute = &matlas.Compute{}
+	regex := regexp.MustCompile("[0-9]+")
+	if d.HasChange("provider_auto_scaling_compute_min_instance_size") {
```

Comment:
Two questions,  why are we only checking if the min is > sizeName and sizeName > max on an update?  The same checks would apply if you set these upon cluster creation?   Did you test if submitting these without checking returned an error code/message from Atlas?

Reply:
Just added validation in create, thank you!!